var/spack/repos/builtin/packages/fftw/package.py

```diff
@@ -202,3 +172,39 @@ def check(self):
 
     def install(self, spec, prefix):
         self.for_each_precision_make('install')
+
+
+class Fftw(FftwBase):
+    """FFTW is a C subroutine library for computing the discrete Fourier
+       transform (DFT) in one or more dimensions, of arbitrary input
+       size, and of both real and complex data (as well as of even/odd
+       data, i.e. the discrete cosine/sine transforms or DCT/DST). We
+       believe that FFTW, which is free software, should become the FFT
+       library of choice for most applications."""
+
+    homepage = "http://www.fftw.org"
+    url = "http://www.fftw.org/fftw-3.3.4.tar.gz"
+    list_url = "http://www.fftw.org/download.html"
+
+    version('3.3.8', sha256='6113262f6e92c5bd474f2875fa1b01054c4ad5040f6b0da7c03c98821d9ae303')
+    version('3.3.7', sha256='3b609b7feba5230e8f6dd8d245ddbefac324c5a6ae4186947670d9ac2cd25573')
+    version('3.3.6-pl2', sha256='a5de35c5c824a78a058ca54278c706cdf3d4abba1c56b63531c2cb05f5d57da2')
+    version('3.3.5', sha256='8ecfe1b04732ec3f5b7d279fdb8efcad536d555f9d1e8fabd027037d45ea8bcf')
+    version('3.3.4', sha256='8f0cde90929bc05587c3368d2f15cd0530a60b8a9912a8e2979a72dbe5af0982')
+    version('2.1.5', sha256='f8057fae1c7df8b99116783ef3e94a6a44518d49c72e2e630c24b689c6022630')
+
+    patch('pfft-3.3.5.patch', when="@3.3.5:+pfft_patches", level=0)
+    patch('pfft-3.3.4.patch', when="@3.3.4+pfft_patches", level=0)
+    patch('pgi-3.3.6-pl2.patch', when="@3.3.6-pl2%pgi", level=0)
+    patch('intel-configure.patch', when="@3:3.3.8%intel", level=0)
+
+    provides('fftw-api@2', when='@2.1.5')
+    provides('fftw-api@3', when='@3:')
+
+    depends_on('automake', type='build', when='+pfft_patches')
+    depends_on('autoconf', type='build', when='+pfft_patches')
+    depends_on('libtool', type='build', when='+pfft_patches')
+
+    variant(
+        'pfft_patches', default=False,
+        description='Add extra transpose functions for PFFT compatibility')
```

Comment:
I usually like to put things in the following order:

1. versions
2. variants
3. dependencies
4. providers
5. patches

Providers aren't very common, so their dependency varies a bit more, but the rest are usually in this order. This isn't a requirement, just a suggestion for consistency with other packages.

Reply:
Reording done